third_party/blink/renderer/core/brave_page_graph/page_graph.cc

```diff
@@ -651,13 +652,31 @@ void PageGraph::RegisterPageGraphScriptCompilation(
           .location_type = classic_script.SourceLocationType(),
       }};
 
+  // Lookup parent script id for kJavascriptUrl scripts.
   if (script_data.source.location_type ==
-      blink::ScriptSourceLocationType::kEvalForScheduledAction) {
-    RegisterScriptCompilationFromEval(execution_context, script_id,
-                                      script_data);
-  } else {
-    RegisterScriptCompilation(execution_context, script_id, script_data);
+          blink::ScriptSourceLocationType::kJavascriptUrl &&
+      script_data.source.parent_script_id == 0) {
```

Comment:
> Would we ever know the parent_script_id for the script_data here? Or is this just to make sure?

this check is here in case Chromium decides to merge `ProcessJavascriptUrl` and `ExecuteJavaScriptUrls` into a single call.